parley/src/layout/editor.rs

```diff
@@ -554,21 +577,40 @@ where
         {
             self.generation.nudge();
         }
-
+        #[cfg(feature = "std")]
+        {
+            let focus = new_sel.focus();
+            let cluster = focus.logical_clusters(&self.layout);
+            let dbg = (
+                cluster[0].as_ref().map(|c| &self.buffer[c.text_range()]),
+                focus.index(),
+                focus.affinity(),
+                cluster[1].as_ref().map(|c| &self.buffer[c.text_range()]),
+            );
+            print!("{dbg:?}");
+            let cluster = focus.visual_clusters(&self.layout);
+            let dbg = (
+                cluster[0].as_ref().map(|c| &self.buffer[c.text_range()]),
+                focus.index(),
+                focus.affinity(),
+                cluster[1].as_ref().map(|c| &self.buffer[c.text_range()]),
+            );
+            println!(" | visual: {dbg:?}");
+        }
         self.selection = new_sel;
     }
 
     /// Get either the contents of the current selection, or the text of the cluster at the caret.
     pub fn active_text(&self) -> ActiveText {
         if self.selection.is_collapsed() {
-            let range = self
-                .selection
-                .focus()
-                .cluster_path()
-                .cluster(&self.layout)
-                .map(|c| c.text_range())
-                .unwrap_or_default();
-            ActiveText::FocusedCluster(self.selection.focus().affinity(), &self.buffer[range])
+            // let range = self
+            //     .selection
+            //     .focus()
+            //     .cluster_path()
+            //     .cluster(&self.layout)
+            //     .map(|c| c.text_range())
+            //     .unwrap_or_default();
+            ActiveText::FocusedCluster(self.selection.focus().affinity(), "")
```

Comment:
:eye: :eye:

Reply:
I have some new debugging code that is more useful so I'll come up with a way to fit that in.

Reply:
I removed `ActiveText` entirely (it was meant for debugging) and replaced the necessary functionality with `PlainEditor::selected_text(&self) -> Option<&str>`

I left my current debugging code in but commented out in `set_selection`. We can clean this up later.

Reply:
I get it now thanks.